File: brainbox-gateway/interfaces/audio.py
from core.logger import setup_logger
from core.config import settings
from httpx import AsyncClient, Timeout, Limits, AsyncHTTPTransport

logger = setup_logger("interfaces/audio.py")

class AudioInterface:
    def __init__(self):
        self.base_url = f"{settings.NEURAL_SERVICE_URL}/audio"
        self.client = AsyncClient(
            timeout=Timeout(10.0),
            limits=Limits(max_connections=100),
            transport=AsyncHTTPTransport(retries=3)
        )

    async def upload_audio(self, headers: dict, file: bytes, filename: str):
        logger.info("Получен /upload-audio запрос для сервиса нейросетей!")
        files = {"file": (filename, file, "audio/*")}
        return await self.client.post(
            f"{self.base_url}/upload-audio",
            headers=headers,
            files=files
        )

    async def recognize_saved_audio(self, headers: dict, audio_uid: str):
        try:
            logger.info("Получен /recognize_saved_audio запрос для сервиса нейросетей!")
            return await self.client.post(
                f"{self.base_url}/recognize-saved-audio",
                headers=headers,
                json={ "audio_uid": audio_uid }
            )
        except Exception as e:
            raise

    async def get_audio_messages(self, headers: dict):
        logger.info("Получен /get-audio-messages запрос для сервиса нейросетей!")
        return await self.client.get(
            f"{self.base_url}/get-audio-messages",
            headers=headers
        )

    async def download_audio(self, headers: dict, audio_uid: str):
        logger.info("Получили /download-audio/{audio_uid} запрос для сервиса нейросетей!")
        return await self.client.get(
            f"{self.base_url}/download-audio/{audio_uid}",
            headers=headers
        )

    async def delete_audio_messages(self, headers: dict):
        logger.info("Получен /delete-audio-messages запрос для сервиса нейросетей!")
        return await self.client.delete(
            f"{self.base_url}/delete-audio-messages",
            headers=headers
        )

audio_interface = AudioInterface()

File: brainbox-client/src/utils/api/audio.js
import axios from "axios";

const GATEWAY_URL_AUDIO = window.appConfig.GATEWAY_URL_AUDIO;

//автоматическая отправка кук
axios.defaults.withCredentials = true;

export const uploadAudio = async (file) => {
  const formData = new FormData();
  formData.append("file", file);
  try {
    const response = await axios.post(
      `${GATEWAY_URL_AUDIO}/upload-audio`,
      formData,
      {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      }
    );
    return response.data;
  } catch (error) {
    console.error("Upload audio error: ", error);
    throw error;
  }
};

export const recognizeSavedAudio = async (audio_uid) => {
  try {
    const response = await axios.post(
      `${GATEWAY_URL_AUDIO}/recognize-saved-audio`,
      {
        audio_uid: audio_uid,
      }
    );
    return response.data;
  } catch (error) {
    console.error(
      "Recognize saved audio error: ",
      error.response?.data || error.message
    );
    throw error;
  }
};

export const checkTaskStatus = async (task_id) => {
  try {
    const response = await axios.get(`${GATEWAY_URL_AUDIO}/tasks/${task_id}`);
    return response.data;
  } catch (error) {
    console.error("Check task status error: ", error);
    throw error;
  }
};

export const getAudioMessages = async () => {
  try {
    console.log("Отправляем запрос на аудио-сообщения");
    const response = await axios.get(
      `${GATEWAY_URL_AUDIO}/get-audio-messages`,
      {
        withCredentials: true,
        headers: {
          "Content-Type": "application/json",
        },
      }
    );
    return {
      messages: response.data.messages.map((msg) => ({
        ...msg,
        table: "audio_chat",
      })),
    };
  } catch (error) {
    console.error("Get audio messages error: ", error);
    throw error;
  }
};

export const downloadAudio = async (audio_uid) => {
  try {
    const response = await axios.get(
      `${GATEWAY_URL_AUDIO}/download-audio/${audio_uid}`,
      {
        responseType: "blob",
        withCredentials: true,
      }
    );
    //создание временной ссылки на скачивание
    const url = window.URL.createObjectURL(new Blob([response.data]));
    const link = document.createElement("a");
    link.href = url;
    link.setAttribute("download", `audio_${audio_uid}.mp3`);
    document.body.appendChild(link);
    link.click();
    link.remove();
    window.URL.revokeObjectURL(url);
  } catch (error) {
    console.error("Download audio error: ", error);
    throw error;
  }
};

export const deleteAudioMessages = async () => {
  try {
    const response = await axios.delete(
      `${GATEWAY_URL_AUDIO}/delete-audio-messages`,
      {
        withCredentials: true,
        headers: {
          "Content-Type": "application/json",
        },
      }
    );
    return response.data;
  } catch (error) {
    console.error("Delete audio messages error: ", error);
    throw error;
  }
};
